ambari-common/src/main/python/resource_management/libraries/functions/service_configs.py

```diff
@@ -0,0 +1,291 @@
+  def get_stack_version(self):
+    return self.get_config_object('hostLevelParams/stack_version')
+
+  @null_key_exception
+  def get_host_level_params(self):
+    return self.get_config_object('hostLevelParams')
+
+  @null_key_exception
+  def get_upgrade_direction(self):
+    return self.get_config_object('commandParams/upgrade_direction')
+
+  @null_key_exception
+  def get_host_sys_preppend(self):
+    return self.get_config_object('hostLevelParams/host_sys_prepped')
+
+  @null_key_exception
+  def get_package_list(self):
+    return self.get_config_object('hostLevelParams/package_list')
+
+  @null_key_exception
+  def check_agent_stack_retry_on_unavailability(self):
+    return bool(self.get_config_object('hostLevelParams/agent_stack_retry_on_unavailability'))
+
+  @null_key_exception
+  def get_agent_stack_retry_count(self):
+    return int(self.get_config_object('hostLevelParams/agent_stack_retry_count'))
+
+  @null_key_exception
+  def get_component_category(self):
+    return self.get_config_object('roleParams/component_category')
+
+  @null_key_exception
+  def get_command_params(self):
+    return self.get_config_object('commandParams')
+
+  @null_key_exception
+  def get_upgrade_type(self):
+    return self.get_config_object('commandParams/upgrade_type')
+
+  @null_key_exception
+  def get_upgrade_direction(self):
+    return self.get_config_object('commandParams/upgrade_direction')
+
+  @null_key_exception
+  def get_service_name(self):
+    return self.get_config_object('serviceName')
+
+  @null_key_exception
+  def get_reconfigure_action(self):
+    return self.get_config_object('commandParams/reconfigureAction')
+
+  @null_key_exception
+  def get_xml_configs_list(self):
+    return self.get_config_object('commandParams/xml_configs_list')
+
+  @null_key_exception
+  def get_env_configs_list(self):
+    return self.get_config_object('commandParams/env_configs_list')
+
+  @null_key_exception
+  def get_properties_configs_list(self):
+    return self.get_config_object('commandParams/properties_configs_list')
+
+  @null_key_exception
+  def get_output_configs_file_name(self):
+    return self.get_config_object('commandParams/output_file')
+
+  @null_key_exception
+  def get_host_name(self):
+    return self.get_config_object('hostname')
+
+  @null_key_exception
+  def check_cluster_security_enabled(self):
+    return self.get_config_object('configurations/clusterSettings/security_enabled')
+
+  @null_key_exception
+  def get_kinit_path(self):
+    return self.get_config_object('configurations/kerberos-env/executable_search_paths')
+
+  @null_key_exception
+  def get_host_java_home(self):
+    return self.get_config_object('hostLevelParams/java_home')
+
+  @null_key_exception
+  def get_host_java_version(self):
+    return self.get_config_object('hostLevelParams/java_version')
+
+  @null_key_exception
+  def get_smokeuser_keytab_path(self):
+    return self.get_config_object('configurations/cluster-settings/smokeuser_keytab')
```

Comment:
cluster-settings is not under configurations but is at the top level.
Same goes with stack-settings.

cluster-env was under configurations :)